pin/pin.go

```diff
@@ -285,32 +288,44 @@ func (p *pinner) Flush() error {
 
 	root := &mdag.Node{}
 	{
-		n, err := storeSet(ctx, p.dserv, p.directPin.GetKeys(), recordInternal)
+		n, err := storeSet(ctx, p.statedag, p.directPin.GetKeys(), recordInternal)
```

Comment:
lol, rebasing 0.4.0 on top of these changes on top of master. lol. thats a real knee slapper there.